utils/zerovec/src/zerovec/mod.rs

```diff
@@ -684,37 +727,56 @@ where
     ///
     /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
     /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
-    /// assert!(matches!(zerovec, ZeroVec::Borrowed(_)));
-    ///
-    /// zerovec.to_mut().push(12_u16.to_unaligned());
-    /// assert!(matches!(zerovec, ZeroVec::Owned(_)));
-    /// ```
-    pub fn to_mut(&mut self) -> &mut Vec<T::ULE> {
-        match self {
-            ZeroVec::Owned(ref mut vec) => vec,
-            ZeroVec::Borrowed(_) => {
-                let vec: Vec<T::ULE> = self.iter().map(T::to_unaligned).collect();
-                let new_self = ZeroVec::Owned(vec);
-                *self = new_self;
-                // recursion is limited since we are guaranteed to hit the Owned branch
-                self.to_mut()
-            }
-        }
+    /// assert!(!zerovec.is_owned());
+    ///
+    /// zerovec.with_mut(|v| v.push(12_u16.to_unaligned()));
+    /// assert!(zerovec.is_owned());
+    /// ```
+    pub fn with_mut<R>(&mut self, f: impl FnOnce(&mut Vec<T::ULE>) -> R) -> R {
```

Comment:
I'm not a fan of the continuation-passing style. Many uses of this actually only need a mutable slice, which is something we can return. Other places, like in datagen, do things like pushing by repeatedly calling `with_mut` which I think is not very readable. They should assemble a Vec and then convert it into a ZeroVec at the end.

Reply:
Good point, added. I didn't move people off of with_mut, this might be a change we can do later